### PR TITLE
chore: improve github actions

### PR DESCRIPTION
**Pull request by koushiro**
Signed-off-by: koushiro <koushiro.cqx@gmail.com>

### CODE REVIEW

**Comment by koushiro** [comment]
@vmx PTAL

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,85 +1,93 @@
...
+          target: wasm32-unknown-unknown
+      - name: Build
+        uses: actions-rs/cargo@v1
         with:
-          args: '--all-features -- --test-threads 1'
-
-      - name: Upload to codecov.io
-        uses: codecov/codecov-action@v1.0.6
+          command: build
+          args: --no-default-features --target wasm32-unknown-unknown
```
Comment (vmx) [inline .github/workflows/build.yml:71]: This one fails for me locally, due to `core` not being there.

Based on https://blog.dbrgn.ch/2019/12/24/testing-for-no-std-compatibility/ `thumbv6m-none-eabi` should work for ensuring `no_std` compatibility.

**Review by vmx — COMMENTED** [review]
This PR removes running clippy with `--no-default-features`, please add that again.

File: .github/workflows/build.yml (hunk 1 above)
Comment (vmx) [reply .github/workflows/build.yml:71]: The code from https://github.com/multiformats/rust-multibase/blob/c87cf2bafbd1e808ed75bb9e7ebf6d80b023fd33/.github/workflows/build.yml#L43-L56 can be used.

**Comment by vmx** [comment]
I've merged https://github.com/multiformats/rust-multihash/pull/97, so you shouldn't need the change to make clippy happy anymore.

**Review by vmx — COMMENTED** [review]
Thanks a lot!

**Pull request merged**